pkg/controller/registry/resolver/step_resolver.go

```diff
@@ -91,7 +91,22 @@ func (r *OperatorStepResolver) ResolveSteps(namespace string) ([]*v1alpha1.Step,
 		return nil, nil, nil, err
 	}
 
-	namespaces := []string{namespace, r.globalCatalogNamespace}
+	// if spec.sourceNamespace is provided on all the subscriptions in the namespace, only consider that namespace for resolution
+	// this prevents errors when unnecessarily searching in the global catalog namespace when those catalogs are unavailable
+	var namespaces = []string{namespace}
+	var searchGlobal bool
+	for _, sub := range subs {
+		if sub.Spec.CatalogSourceNamespace == namespace {
+			continue
+		} else {
+			searchGlobal = true
+			break
+		}
+	}
+	if searchGlobal {
```

Comment:
I feel this namespace exclusion approach is a bit too broad though I can see it is the simplest solution for this particular problem. This solution preemptively excludes global catalogsources from resolution regardless if there are any issues with them or not. I would like to see if we can have a more refined solution that will still consider global catalogsources and simply ignore errors from those catalogsources if occur and let the resolution continue.

Reply:
Ignoring catalog errors breaks resolution determinism, since any transient error could cause a different result. How about making this explicitly configurable on a per-namespace basis via a new field on the OperatorGroup spec?

Reply:
Adding configurable settings in OperatorGroup is quite trending later so I don't see the harm of doing the same for this particular scenario. My point still stands though on the fact preemptive solution is too broad.

Reply:
> Ignoring catalog errors breaks resolution determinism, since any transient error could cause a different result. How about making this explicitly configurable on a per-namespace basis via a new field on the OperatorGroup spec?

@benluddy I think we have other options too, like:
- lean on the SAT-solver for resolution determinism; e.g. operators that require deterministic resolution have constraints on unique label properties only available in the "correct catalog"